### PR TITLE
[WIP] Fix compiler warnings

### PR DESCRIPTION
**Pull request by rapperskull**
Some of them are just annoying, while others highlight real problems. Still not finished, has to be tested.

### CODE REVIEW

**Comment by Extrems** [comment]
I'm not digging all the PATHNAME_MAX math. Should be using snprintf though.

**Comment by rapperskull** [comment]
Yes and no. `snprintf` is indeed more clear, but produces another warning. I chose this approach for two reasons: it truly avoids warnings without having to suppress them and also truncates only the first part of the path, so that it's more likely that the resulting path is non-existent. The "true" solution would be using `snprintf`, actually checking for truncation and aborting when needed.

**Comment by rapperskull** [comment]
Now only four warnings remain, making real problems easier to spot.
One is related to `fatfs` and should be solved from their side.
Of the other three, one is a potentially unaligned pointer, while the other two are writes beyond the end of the buffer.

**Comment by Extrems** [comment]
I'd rather dismiss the warnings than doing this nonsense. This is far too messy.

**Comment by rapperskull** [comment]
It depends on what warnings you're talking about. For the ones generated by `snprintf`, I agree, even if it's not the optimal solution.
The best solution would be to actually handle truncation, but I understand it doesn't make sense in most cases.

**Comment by rapperskull** [comment]
What do you think of this solution?

File: cube/swiss/source/config/config.c [hunk 1]
```diff
@@ -51,8 +51,7 @@ static int configEntriesCount = 0;
 
 
 void strnscpy(char *s1, char *s2, int num) {
-	strncpy(s1, s2, num);
-	s1[num] = 0;
+	sprintf(s1, "%.*s", num, s2);
```
Comment (Extrems) [inline cube/swiss/source/config/config.c:54]: We have strlcpy. Shouldn't this be used here?

Comment (rapperskull) [reply cube/swiss/source/config/config.c:54]: You're right.

**Pull request merged**